MM-T399 Invalidate Pending Email Invitations

josephk96 opened the pull request:
I created a test to check that invites a user to a team, ensures the user can sign up via the email link provided, invite two more users, invalidate their invites, retrieve the email of the second user and go to the provided sign up link and validate that the correct error message is shown.

Test case: https://automation-test-cases.vercel.app/test/MM-T399

saturninoabril commented on e2e/cypress/integration/onboarding/invalidate_pending_email_invitations_spec.js:
```diff
@@ -0,0 +1,172 @@
+
+        // # Invite the first user (the one that won't be invalidated and logout)
+        inviteNewUser(emailOne);
+        cy.wait(TIMEOUTS.FIVE_SEC);
```
I think it should click "Done" to complete the process.
```suggestion
        cy.findByText('Done').should('be.visible').click();
```

saturninoabril commented on e2e/cypress/integration/onboarding/invalidate_pending_email_invitations_spec.js:
```diff
@@ -0,0 +1,172 @@
+        cy.get('#createAccountButton').click();
+
+        // * Check that the display name of the team the user was invited to is being correctly displayed
+        cy.get('#headerTeamName').should('contain.text', testTeam.display_name);
```
I'm getting random failing test here which can be solved by adding a timeout option.
```suggestion
        cy.get('#headerTeamName', {timeout: TIMEOUTS.HALF_MIN}).should('contain.text', testTeam.display_name);
```

saturninoabril commented on e2e/cypress/integration/onboarding/invalidate_pending_email_invitations_spec.js:
```diff
@@ -0,0 +1,172 @@
+
+        // # Logout from the current user and login as sysadmin
+        cy.apiLogout();
+        cy.apiAdminLogin();
```
I think it needs to reload a page after login.

saturninoabril commented on e2e/cypress/integration/onboarding/invalidate_pending_email_invitations_spec.js:
```diff
@@ -0,0 +1,172 @@
+        cy.apiLogout();
+        cy.apiAdminLogin();
+
+        // # Open the 'Invite People' full screen modal
+        cy.get('.sidebar-header-dropdown__icon').click();
+        cy.get('#invitePeople').find('button').eq(0).click();
+
+        // # Wait half a second to ensure that the modal has been fully loaded and check if the instance is licensed
+        cy.wait(TIMEOUTS.HALF_SEC);
+        checkIfLicensed();
```
```suggestion
        // # Open the 'Invite People' full screen modal
        cy.findByLabelText('main menu').should('be.visible').click();
        cy.findByText('Invite People').should('be.visible').click();
```

saturninoabril reviewed (changes requested):
Thanks @josephk96! Please see comments.

josephk96 commented:
@saturninoabril ,  I updated the `team_settings/helpers.js` file to be in sync with https://github.com/mattermost/mattermost-webapp/pull/6263/files and used the methods there in this test. I hope that it would be fine, since I am in the `onboarding` directory but am importing and using functions from another directory. If it isn't, please let me know and I will refactor this. I also added the signup page visibility check in the verifyEmailInviteAndVisitLink function :)

@josephbaylon, please let me know if the test is still failing for you :)

josephbaylon commented:
/update-branch

josephk96 commented:
I also tested MM-T389 and MM-T386 to use the `verifyEmailInviteAndVisitLink` function with the newly included signup section visibility check and they are working

josephk96 commented:
Thanks @josephbaylon ! Btw, I'm a big fan of your name, it's super cool💯

josephbaylon commented:
@josephk96 Hahaha. Likewise 🤣

saturninoabril reviewed (approved):
Thanks @josephk96!
